### PR TITLE
fix: Too long identifier name error

### PR DESCRIPTION
**Pull request by MaxymVlasov**
## what

Fix too long identifier name without cluster recreation:

set `id_length_limit = 40` for

```tf
module "aurora" {
  source  = "cloudposse/rds-cluster/aws"
  version = "1.10.0"
  ...
  id_length_limit = 40
}
``` 
will cause such recreation if final `module.this.id` will be between 40-60 characters

## why

https://github.com/cloudposse/terraform-aws-rds-cluster/releases/tag/1.10.0 introduced `random_pet`, which adds 2 extra words on the top of `module.this.id`, and that exceed limit of 63 allowed characters for RDS names

```tf
╷
│ Error: creating RDS Cluster (company-staging-aurora-payments-ledger-service-aliases) Instance (company-staging-aurora-payments-ledger-service-aliases-promoted-piglet-1): operation error RDS: CreateDBInstance, https response error StatusCode: 400, RequestID: 1979b42f-b1df-4e00-b0ec-d3b629b3002d, api error InvalidParameterValue: Invalid database identifier:  company-staging-aurora-payments-ledger-service-aliases-promoted-piglet-1
│ 
│   with module.aurora_aliases.aws_rds_cluster_instance.default[0],
│   on .terraform/modules/aurora_aliases/main.tf line 261, in resource "aws_rds_cluster_instance" "default":
│  261: resource "aws_rds_cluster_instance" "default" {
```

This PR limit final identifier to 62-63 chars (depends on count of replica)

## references

Fixing https://github.com/cloudposse/terraform-aws-rds-cluster/pull/213

### CODE REVIEW

**Comment by kevcube** [comment]
@MaxymVlasov we still want to retain that random_pet because when we `create_before_destroy` we need to have unique names for the new instances.

**Comment by MaxymVlasov** [comment]
@kevcube 
TL;DR: `id_length_limit` in `cloudposse/label/null` calculate `md5()` of full_id, crop full_id and attach md5 at the end.

So you still have unique id based on pet_random, but which not violate AWS limitation

https://github.com/cloudposse/terraform-null-label/blob/5c464f42c388deb18222c2285c9e450b1fc233c7/main.tf#L161-L166

File: main.tf [hunk 1]
```diff
@@ -268,9 +268,28 @@ resource "random_pet" "instance" {
...
+  # Setting the limit to 60 allow to use up to 99 instances, when only 16 is allowed
+  # (1 writer + 15 readers)
+  # https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/Aurora.Replication.html
+  id_length_limit = 60
+}
+
+output "cropped_id" {
+  value = module.rds_identifier[0].id
+}
+
```
Comment (MaxymVlasov) [inline main.tf:289]:
```suggestion
```

**Review by kevcube — APPROVED** [review]
Ah I see, I only looked at the part where you replace `random_pet` with `module.rds_identifier` and I assumed that was an identifier that was already being created, now I see it is a new one. LGTM

File: main.tf [hunk 2]
```diff
@@ -268,9 +268,24 @@ resource "random_pet" "instance" {
   }
 }
 
+module "rds_identifier" {
+  count  = local.enabled ? 1 : 0
```
Comment (kevcube) [inline main.tf:272]:
```suggestion
  count = local.enabled ? 1 : 0
```

**Review by kevcube — APPROVED** [review]

**Comment by MaxymVlasov** [comment]
@kevcube tests passed

**Comment by kevcube** [comment]
> @kevcube tests passed

ok - one thing I'm wondering is how we release this, it feels like a fix, but also this will cause consumer's db instances to be recycled so maybe I release as a minor. Going to ask for backup.

**Comment by MaxymVlasov** [comment]
It will affect only DBs with a `random_pet.id` length == 61 (>=62 will see the same error as in PR description)
I don't know, how many DBs have exactly this number, so yeah, second eyes could be helpful

**Comment by kevcube** [comment]
Although I cannot merge because I was the last pusher and also only approver 🤡 whoops!
